backend/src/main/java/ca/mcgill/purposeful/controller/AppUserController.java

```diff
@@ -81,4 +90,123 @@ public ResponseEntity<AppUserDto> registerModerator(@RequestBody AppUserDto appU
+          appUserService.modifyUserNames(email, firstname, lastname));
+  
+      return ResponseEntity.status(HttpStatus.OK).body(registeredUser);
+    }
+
+    /**
+     * PUT method to update the password of a regular user
+     * @param appUserDto - the user to modify the password
+     * @return the modified user
+     * 
+     * @author Enzo Benoit-Jeannin
+     */
+    @PutMapping(value = {"/regular/password",
+      "/regular/password/"}, consumes = "application/json", produces = "application/json")
+    @PreAuthorize("hasAnyAuthority('Owner', 'Moderator', 'User')")
+    public ResponseEntity<AppUserDto> updatePasswordRegularUser(@RequestBody AppUserDto appUserDto) {
+      // Unpack the DTO
+      if (appUserDto == null) {
+        throw new GlobalException(HttpStatus.BAD_REQUEST, "AppUserDto is null");
+      }
+
+      String email = appUserDto.getEmail();
+      String password = appUserDto.getPassword();
+  
+      // Register the user
+      AppUserDto registeredUser = DtoUtility.convertToDto(
+          appUserService.modifyPassword(email, password));
+  
+      return ResponseEntity.status(HttpStatus.OK).body(registeredUser);
+    }
+
+  /**
+   * GET method to get all users
+   * 
+   * @return a list of all users
+   * 
+   * @author Enzo Benoit-Jeannin
+   */
+  @GetMapping(value = {"/users", "/users/"})
+  public List<AppUserDto> getAllUsers() {
+    return appUserService.getAllUsers().stream()
+        .map(e -> DtoUtility.convertToDto(e))
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * PUT method to update the names of a moderator
+   * @param appUserDto - the moderator to modify the names
+   * @return the modified moderator
+   * 
+   * @author Enzo Benoit-Jeannin
+   */
+  @PutMapping(value = {"/moderator",
+  "/moderator/"}, consumes = "application/json", produces = "application/json")
+  @PreAuthorize("hasAnyAuthority('Owner', 'Moderator')")
+  public ResponseEntity<AppUserDto> updateModerator(@RequestBody AppUserDto appUserDto) {
+    // Unpack the DTO
+    if (appUserDto == null) {
+      throw new GlobalException(HttpStatus.BAD_REQUEST, "AppUserDto is null");
+    }
+    String email = appUserDto.getEmail();
+    String firstname = appUserDto.getFirstname();
+    String lastname = appUserDto.getLastname();
+
+    // Register the user
+    AppUserDto registeredUser = DtoUtility.convertToDto(
+        moderatorService.modifyModerator(email, firstname, lastname));
+
+    return ResponseEntity.status(HttpStatus.OK).body(registeredUser);
+  }
+
+  /**
+     * PUT method to update the password of a moderator
+     * @param appUserDto - the moderator to modify the password
+     * @return the modified moderator
+     * 
+     * @author Enzo Benoit-Jeannin
+     */
+    @PutMapping(value = {"/mdoerator/password",
+      "/moderator/password/"}, consumes = "application/json", produces = "application/json")
+    @PreAuthorize("hasAnyAuthority('Owner', 'Moderator')")
```

Comment:
Same comment here : 
The endpoint should not be available to all moderators, only the concerned moderator and the owners should be able to modify it. Even then, I am not sure if the owners should be able to modify this

Reply:
I added the restriction in my last pull request (the step definitions pull request)

Reply:
Very nice!